concordium-std/src/types.rs

```diff
@@ -581,6 +584,93 @@ pub struct ExternReturnValue {
     pub(crate) current_position: u32,
 }
 
+/// The current public balances of an account.
+#[derive(Debug, Copy, Clone)]
+pub struct AccountBalance {
+    /// The total balance of the account. Note that part of this balance might
+    /// be staked and/or locked in scheduled transfers.
+    pub(crate) total:  Amount,
+    /// The current staked amount of the account. This amount is used for
+    /// staking.
+    pub(crate) staked: Amount,
+    /// The current amount locked in releases that resulted from transfers with
+    /// schedule. A locked amount can still be used for staking.
+    pub(crate) locked: Amount,
+}
+
+impl AccountBalance {
+    /// Construct a new account balance.
+    pub fn new(total: Amount, staked: Amount, locked: Amount) -> Self {
```

Comment:
We could check that `total >= max(locked, staked)`

Reply:
I have added a check for this now